### PR TITLE
cleanup(docs): update readme + include more verbose testing instructions

### PR DESCRIPTION
**Pull request by incertum**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

/kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap-engine-udig

> /area libscap

> /area libpman

> /area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

Revamp and clarify the `libs` README to make it more technically accessible and to provide a smoother onboarding experience for new contributors.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -173,46 +192,98 @@ To build the modern eBPF probe, you need:
     ```
 * A kernel version >=`5.8`.
 
-Then, issue:
+> __NOTE:__ These are not the requirements to use the modern BPF probe, but rather for building it from source.
```
Comment (incertum) [inline README.md:195]: @Andreagit97 are these changes ok? Would you have additional improvement suggestions? Thanks in advance!

**Comment by incertum** [comment]
@mrgian and @oheifetz may I kindly ask for your feedback re the proposed changes https://github.com/falcosecurity/libs/blob/0f041148624e1c3c1c401fa58a04cae0e4897571/README.md? Would it have been easier to onboard as new developer with these updates? do you have additional suggestions? Thank you!

**Comment by mrgian** [comment]
Yeah I think new developers need to know how to run tests locally on their machines.
Before this I had to check failed CI tests to know which commands to run.

**Comment by oheifetz** [comment]
Hi Melissa,

I think it would be more readable if the cmake configuration would be multi
line and if there was a oneliner per CONFIG it would be even better.
Also specify what is the default value for a flag, can one assume that
default value is the opposite to the value used in the cmake example?

regarding -j, you can put -j$(($nproc)-1)), that way a user leaves one core
out of build, so they will not OOM
also missing is the actual run of drivers_test -m/b/k, I think

Also can suggests to use ccache for improved build time, especially for
developers who build repeatedly

-Ofer

File: README.md (hunk 1 above)
Comment (Andreagit97) [reply README.md:195]: Super detailed, thanks :+1: !

File: README.md [hunk 2]
```diff
@@ -173,46 +192,98 @@ To build the modern eBPF probe, you need:
...
+rm -f driver/src/scap.ko; make driver;
+rm -f libscap/examples/01-open/scap-open; make -j 16 scap-open;
+rm -f libsinsp/examples/sinsp-example; make -j 16 sinsp-example;
+```
+
+These are the conventional unit tests that our CI system enforces:
+
+```bash
+make -j 16 unit-test-libsinsp; make run-unit-test-libsinsp;
+make -j 16 run-unit-test-libscap; sudo ./libscap/test/unit-test-libscap;
```
Comment (Andreagit97) [inline README.md:260]: i think we don't this anymore, all these tests have been ported to `make linscap_test`

**Comment by incertum** [comment]
@FedeDP you are back, would you have additional feedback on what we might need to make it more onboarding friendly?

**Comment by FedeDP** [comment]
I love it @incertum ! I think you already covered everything that was needed!
Thank you very much for this effort.

**Review by FedeDP — APPROVED** [review]
/approve

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**